pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -99,7 +100,7 @@ private long getCutoffTime() {
     }
 
     @Override
-    public boolean addMessage(long ledgerId, long entryId, long deliverAt) {
+    public synchronized boolean addMessage(long ledgerId, long entryId, long deliverAt) {
```

Comment:
I think it is best to use the same lock as `getScheduledMessages`. Keep use `synchronized(dispatcher)`?

Reply:
In my opinion it is better to ensure thread safety by `DelayedDeliveryTracker` itself

Reply:
Ok, also add `synchronized` for the others method in the  `InMemoryDelayedDeliveryTracker`.

Reply:
I don't think we need `synchronized` for all method in `InMemoryDelayedDeliveryTracker` though, we're not doing a big refactor here and just trying to avoid unnecessary synchronization if delayedDeliveryTracker already present right